src/ImageSharp/Formats/Jpeg/Components/Encoder/HuffmanScanEncoder.cs

```diff
@@ -388,5 +388,40 @@ private void FlushInternalBuffer()
                 this.target.Write(this.emitBuffer, 0, this.emitLen);
             }
         }
+
+        /// <summary>
+        /// Calculates how many minimum bits needed to store given value for Huffman jpeg encoding.
+        /// </summary>
+        /// <remarks>
+        /// This method returns 0 for input value 0. This is done specificaly for huffman encoding
+        /// </remarks>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int GetHuffmanEncodingLength(uint value)
+        {
+            DebugGuard.IsTrue(value <= (1 << 16), "Huffman encoder is supposed to encode a value of 16bit size max");
+#if SUPPORTS_BITOPERATIONS
+            // This should have been implemented as (BitOperations.Log2(value) + 1) as in non-intrinsic implementation
+            // But internal log2 is implementated like this: (31 - (int)Lzcnt.LeadingZeroCount(value))
+
+            // BitOperations.Log2 implementation also checks if input value is zero for the convention 0->0
+            // Lzcnt would return 32 for input value of 0 - no need to check that with branching
+            // Fallback code if Lzcnt is not supported still use if-check
+            // But most modern CPUs support this instruction so this should not be a problem
+            return 32 - System.Numerics.BitOperations.LeadingZeroCount(value);
+#else
+            // Ideally:
+            // if 0 - return 0 in this case
+            // else - return log2(value) + 1
+            //
+            // Hack based on input value constaint:
+            // We know that input values are guaranteed to be maximum 16 bit large for huffman encoding
+            // We can safely shift input value for one bit -> log2(value << 1)
+            // Because of the 16 bit value constraint it won't overflow
+            // With that input value change we no longer need to add 1 before returning
+            // And this eliminates need to check if input value is zero - it is a standard convention which Log2SoftwareFallback adheres to
+            return Numerics.Log2SoftwareFallback(value << 1);
```

Comment:
```suggestion
            return Numerics.Log2(value << 1);
```

Reply:
`Log2SoftwareFallback` should be private.

Reply:
LGTM!
I'll also mark Log2SoftwareFallback as private cause jpeg encoder is the only thing that needed access to it, everything else should call Log2 polyfill.